accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/resource/ReportController.java

```diff
@@ -118,4 +119,16 @@ public ResponseEntity<ReportResponseView> reportPublish(@Valid @RequestBody Repo
         );
     }
 
+    @Tag(name = "Reporting", description = "Public search for reporting")
+    @PostMapping(value = "/public/report-list", produces = "application/json")
+    public ResponseEntity<ReportResponseView> reportSearchPublicInterface(@Valid @RequestBody PublicReportSearchRequest reportSearchRequest) {
+
+        return ResponseEntity.ok().body(ReportResponseView.createSuccess(reportService.findAllByTypeAndPeriod(
+                        reportSearchRequest.getReportType(),
+                        reportSearchRequest.getIntervalType(),
+                        reportSearchRequest.getYear(),
+                        reportSearchRequest.getPeriod()
+                ).stream().map(reportViewService::responseView).collect(Collectors.toSet()))
```

Comment:
You may consider toList() but it is up to you if we need this, generally reports should be deduped anyway